transform/transformers/PDFTransformer.py

```diff
@@ -67,14 +55,20 @@ def __init__(self, survey, response_data):
         self.response = response_data
 
     def render(self):
+        """Get the pdf data in memory"""
+        return self.render_pages()[0]
+
+    def render_pages(self):
+        """Return both the in memory pdf data and a count of the pages"""
         buffer = BytesIO()
         doc = SimpleDocTemplate(buffer, pagesize=A4)
         doc.build(self.get_elements())
 
         pdf = buffer.getvalue()
+
         buffer.close()
 
-        return pdf
+        return pdf, doc.page
 
     def render_to_file(self):
```

Comment:
This method still writes to the File system?

Reply:
Yep ,its not used . Removing .